include/deal.II/fe/mapping_q_eulerian.h

```diff
@@ -200,6 +201,14 @@ class MappingQEulerian : public MappingQ<dim, spacedim>
     std::vector<Point<spacedim> >
     compute_mapping_support_points(const typename Triangulation<dim,spacedim>::cell_iterator &cell) const;
 
+    /**
+     * Always returns @p false because MappingQ1Eulerian does not in general
+     * preserve vertex locations (unless the translation vector happens to
+     * provide for zero displacements at vertex locations).
+     */
+    virtual
+    bool preserves_vertex_locations () const;
```

Comment:
The function in line 132 is for the class `MappingQEulerian`, whereas the new declaration that was previously missing (and resulted in wrong results) is for the class `MappingQEulerian::MappingQEulerianGeneric`, note the class declared inside the class in line 175. We need this double definition because `MappingQEulerian` derives from `MappingQ` and thus needs to define separate operations for the near-boundary region and the interior. This can be simplified once we have done #4462.